backend/fms_core/services/container.py

```diff
@@ -0,0 +1,157 @@
+                errors.append(f"Parent container kind {container_parent.kind} requires that you provide coordinates.")
+            else:
+                try:
+                    container = Container.objects.create(**container_data, comment=comment)
+                    created_entity = True
+                # Pile up all validation error raised during the creation of the container
+                except ValidationError as e:
+                    errors.append(';'.join(e.messages))
+    else:
+        errors.append(f"Barcode is required to get or create a container.")
+
+    return (container, created_entity, errors, warnings)
+
+def create_container(barcode, kind,
+                     name=None, coordinates=None, container_parent=None, creation_comment=None):
+    container = None
+    errors = []
+    warnings = []
+
+    if barcode:
+        if Container.objects.filter(barcode=barcode).exists():
+            errors.append(f"Container with barcode {barcode} already exists.")
+        else:
+            container_data = dict(
+                **(dict(location=container_parent) if container_parent else dict()),
+                **(dict(barcode=barcode) if barcode is not None else dict()),
+                **(dict(name=name) if name is not None else dict(name=barcode)), # By default, a container name will be his barcode
+                **(dict(coordinates=coordinates) if coordinates is not None else dict()),
+                **(dict(kind=kind) if kind is not None else dict()),
+            )
+            comment = creation_comment or (f"Automatically generated on {datetime.utcnow().isoformat()}Z")
+
+            if container_parent and CONTAINER_KIND_SPECS[container_parent.kind].requires_coordinates and not coordinates:
+                errors.append(f"Parent container kind {container_parent.kind} requires that you provide coordinates.")
+            else:
+                try:
+                    container= Container.objects.create(**container_data, comment=comment)
+
+                # Pile up all validation error raised during the creation of the container
+                except ValidationError as e:
+                    errors.append(';'.join(e.messages))
+    else:
+        errors.append(f"Barcode is required to create a container.")
+
+    return (container, errors, warnings)
+
+def rename_container(container_to_update, barcode=None, name=None, update_comment=None):
+    errors = []
+    warnings = []
+
+    if not any([barcode, name]):
+        errors.append(f'Either New Barcode or New Name are required.')
+        return (container_to_update, errors, warnings)
+
+    if barcode:
+        container_to_update.barcode = barcode
+    if name:
+        container_to_update.name = name
+    if update_comment:
+        container_to_update.update_comment = update_comment
+
+    try:
+        container_to_update.save()
+    except Exception as e:
+        errors.append(str(e))
+
+    return (container_to_update, errors, warnings)
```

Comment:
Do we need to return the container ... this appear a useless return value. Unless you create a copy of the object you are basically giving back the parameter reference... Same applies to move